tools/progen_compile.py

```diff
@@ -50,14 +50,16 @@ def load_project_list(path):
     return project_list
 
 
-def build_projects(file, projects, tool, build=True, clean=False, ignore_errors=True):
+def build_projects(file, projects, tool, build=True, clean=False, parallel=False, ignore_failures=False, verbose=False):
     generator = generate.Generator(file)
-    cores = 4
-    try:
-        import multiprocessing
-        cores = multiprocessing.cpu_count()
-    except:
-        pass
+    cores = 1
+    if parallel:
+        cores = 4
+        try:
+            import multiprocessing
+            cores = multiprocessing.cpu_count()
+        except:
+            pass
```

Comment:
You don't need to change it, but it would be more Pythonic to move the `cores = 4` to the `except` clause.